tests/rados/test_osd_full.py

```diff
@@ -694,3 +674,157 @@ def check_full_warning(type, a_set):
+
+        try:
+            rados_obj.configure_pg_autoscaler(**{"default": "off"})
+            pool1_name = pool_name = config["pool-1"]["pool_name"]
+            assert create_pool_per_config(**config["pool-1"])
+
+            # retrieving the size of each osd part of acting set for the pool
+            acting_set1 = rados_obj.get_pg_acting_set(pool_name=pool1_name)
+            match_osd_size(a_set=acting_set1)
+            primary_osd_size = osd_sizes[acting_set1[0]]
+
+            log.info(
+                f"All OSDs part of acting set {acting_set1} are of same size {primary_osd_size} KB"
+            )
+
+            # reweight osds to 0
+            [
+                cephadm.shell([f"ceph osd reweight osd.{osd_id} 0"])
+                for osd_id in acting_set1
+            ]
+
+            pool2_name = config["pool-2"]["pool_name"]
+            assert create_pool_per_config(**config["pool-2"])
+            acting_set2 = rados_obj.get_pg_acting_set(pool_name=pool2_name)
+
+            pool3_name = config["pool-3"]["pool_name"]
+            assert create_pool_per_config(**config["pool-3"])
+            acting_set3 = rados_obj.get_pg_acting_set(pool_name=pool3_name)
+
+            for osd_id in acting_set1:
+                assert osd_id not in acting_set2
+                assert osd_id not in acting_set3
+
+            log.info(
+                "None of the OSDs part of acting set of Pool 1 are part of acting set of Pool 2 and 3"
+            )
+            log.info(
+                f"\n Acting set of Pool 1: {acting_set1} | Acting set of Pool 2: {acting_set2} | "
+                f"Acting set of Pool 3: {acting_set3}"
+            )
+
+            # change the nearfull, backfill-full and full-ratio to 0.60, 0.65, and 0.7 respectively
+            cmds = [
+                "ceph osd set-nearfull-ratio 0.65",
+                "ceph osd set-backfillfull-ratio 0.69",
+                "ceph osd set-full-ratio 0.70",
+            ]
+            [cephadm.shell(args=[cmd]) for cmd in cmds]
+
+            # determine the number of objects to be written to the pool
+            # to achieve nearfull, backfillfull and full state.
+            full_objs = int(primary_osd_size * 0.7 / bench_obj_size_kb) + 1
+
+            # reweight osds to 1
+            [
+                cephadm.shell([f"ceph osd reweight osd.{osd_id} 1"])
+                for osd_id in acting_set1
+            ]
+
+            # perform rados bench to trigger full warning
+            full_config = {
+                "seconds": 200,
+                "b": f"{bench_obj_size_kb}KB",
+                "no-cleanup": True,
+                "max-objects": full_objs,
+            }
+            bench_obj.write(client=client_node, pool_name=pool1_name, **full_config)
+
+            log.info(f"{full_objs + 1} objects have been written to pool {pool1_name}")
+
+            check_full_warning(type="full", a_set=acting_set1)
+
+            # perform rados bench to trigger full warning
```

Comment:
The rados bench on pool2 and pool3 is not triggering full warning but to verify IOPS are possible. Rewording is needed here.

Reply:
Sorry, missed it, fixed now